src/mysql_protocol.erl

```diff
@@ -42,6 +42,8 @@
 -define(ok_pattern, <<?OK, _/binary>>).
 -define(error_pattern, <<?ERROR, _/binary>>).
 -define(eof_pattern, <<?EOF, _:4/binary>>).
+-define(fast_auth_success_pattern, <<1:8, ?SHA2_OK:8>>).
+-define(full_auth_pattern, <<1:8, ?SHA2_FULL_AUTH:8>>).
```

Comment:
Does `1` have a meaning on its own or is it just the combinations `<<1, 3>>` and `<<1, 4>>` that have a meaning?

To me, it looks like `1` is the packet tag for "sha2_auth_read" and the next byte is the payload. What do you think about a special parse function for this packet type? Something like this one:

```Erlang
parse_sha2_auth_read_packet(<<?sha2_auth_read:8, Msg:8>>) ->
    #sha_auth_read{msg = Msg}.
```